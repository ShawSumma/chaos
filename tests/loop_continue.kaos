num def f1()
    num a = 101
    return a
end

num def f2()
    num b = 102
    return b
end

num def f3()
    num c = 103
    return c
end

num def f4(num x, num y)
    num z = x + y
    bool a = true
end {
    z == 5  : continue,
    z == 8  : f1(),
    z > 10  : f2(),
    default : f3()
}

num def f5(num x, num y)
    num z = x + y
    bool a = true
end {
    z == 5  : f3(),
    z == 8  : f1(),
    z > 10  : f2(),
    default : continue
}

list a = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15]

foreach a as el
    print f4(el, 1)
    print el
end

print "hey"

foreach a as el
    print f5(el, 1)
    print el
end

print "hoy"
