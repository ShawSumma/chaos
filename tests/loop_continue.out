103
1
103
2
103
3
103
5
103
6
101
7
103
8
103
9
102
10
102
11
102
12
102
13
102
14
102
15
hey
103
4
101
7
102
10
102
11
102
12
102
13
102
14
102
15
hoy
